### PR TITLE
Allowing more than 3/4 dims for `total_variation` + adding `reduction`

### PR DESCRIPTION
**Pull request by nitaifingerhut**
#### Changes
Improvements to the `total_variation` loss.

#### Type of change

- [x] 📚  Documentation Update
- [x] 🔬 New feature (non-breaking change which adds functionality)

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

**Review by edgarriba — COMMENTED** [review]
Can you add quick tests ? Would be great also to upgrade the function style by using Kornia.core Tensor and reuse the KORNIA_CHECK api

**Comment by nitaifingerhut** [comment]
@edgarriba  fixed your comments

**Comment by nitaifingerhut** [comment]
The pre-commit.ci failed due to unused import of `torch`, but this is used in the examples in the docs.

File: kornia/losses/total_variation.py [hunk 1]
```diff
@@ -22,18 +28,23 @@ def total_variation(img: torch.Tensor) -> torch.Tensor:
     Reference:
         [1] https://en.wikipedia.org/wiki/Total_variation
     """
-    if not isinstance(img, torch.Tensor):
-        raise TypeError(f"Input type is not a torch.Tensor. Got {type(img)}")
-
-    if len(img.shape) < 3 or len(img.shape) > 4:
-        raise ValueError(f"Expected input tensor to be of ndim 3 or 4, but got {len(img.shape)}.")
+    KORNIA_CHECK_IS_TENSOR(img)
```
Comment (edgarriba) [inline kornia/losses/total_variation.py:31]: check shape already check tensor

File: kornia/losses/total_variation.py [hunk 2]
```diff
@@ -1,15 +1,21 @@
 import torch
```
Comment (edgarriba) [inline kornia/losses/total_variation.py:1]: you can remove torch imports

Comment (edgarriba) [reply kornia/losses/total_variation.py:1]: the module is exposed also in kornia.core

File: kornia/losses/total_variation.py [hunk 3]
```diff
@@ -22,18 +29,22 @@ def total_variation(img: torch.Tensor) -> torch.Tensor:
...
     pixel_dif2 = img[..., :, 1:] - img[..., :, :-1]
 
-    reduce_axes = (-3, -2, -1)
-    res1 = pixel_dif1.abs().sum(dim=reduce_axes)
-    res2 = pixel_dif2.abs().sum(dim=reduce_axes)
+    res1 = pixel_dif1.abs()
+    res2 = pixel_dif2.abs()
+
+    reduce_axes = (-2, -1)
+    if reduction == "mean":
```
Comment (edgarriba) [inline kornia/losses/total_variation.py:42]: something i'm thinking now, should we support also `none` ?

File: kornia/losses/total_variation.py [hunk 4]
```diff
@@ -1,19 +1,26 @@
-import torch
 import torch.nn as nn
 
+from kornia.core import Tensor
```
Comment (edgarriba) [inline kornia/losses/total_variation.py:3]:
```suggestion
from kornia.core import Tensor, Module
```

File: kornia/losses/total_variation.py [hunk 5]
```diff
@@ -1,19 +1,26 @@
-import torch
 import torch.nn as nn
```
Comment (edgarriba) [inline kornia/losses/total_variation.py:1]:
```suggestion
```

File: kornia/losses/total_variation.py (hunk 3 above)
Comment (nitaifingerhut) [reply kornia/losses/total_variation.py:42]: I think no. 
1. `res1` and `res2` have different shapes; (H-1, W) and (H, W-1)
2. commonly TV is averages/summed.

File: kornia/losses/total_variation.py [hunk 6]
```diff
@@ -1,19 +1,26 @@
...
 
+from kornia.core import Tensor
+from kornia.testing import KORNIA_CHECK, KORNIA_CHECK_SHAPE
 
-def total_variation(img: torch.Tensor) -> torch.Tensor:
+
+def total_variation(img: Tensor, reduction: str = "sum") -> Tensor:
```
Comment (edgarriba) [inline kornia/losses/total_variation.py:7]: any strong opinion why `sum` by default ? i think we default to reduction `none` in some places. Just to standardize over the lib

**Review by edgarriba — COMMENTED** [review]
few more comments

File: kornia/losses/total_variation.py (hunk 6 above)
Comment (nitaifingerhut) [reply kornia/losses/total_variation.py:7]: to be backward compatible

File: kornia/losses/total_variation.py (hunk 3 above)
Comment (edgarriba) [reply kornia/losses/total_variation.py:40]: oh right. Nevermind

File: kornia/losses/total_variation.py (hunk 6 above)
Comment (edgarriba) [reply kornia/losses/total_variation.py:5]: fair enough

**Comment by nitaifingerhut** [comment]
lass `TotalVariation(nn.Module)` should be lass `TotalVariation(Module)`

**Review by edgarriba — APPROVED** [review]

File: kornia/losses/total_variation.py [hunk 7]
```diff
@@ -22,39 +27,44 @@ def total_variation(img: torch.Tensor) -> torch.Tensor:
...
-    reduce_axes = (-3, -2, -1)
-    res1 = pixel_dif1.abs().sum(dim=reduce_axes)
-    res2 = pixel_dif2.abs().sum(dim=reduce_axes)
+    res1 = pixel_dif1.abs()
+    res2 = pixel_dif2.abs()
+
+    reduce_axes = (-2, -1)
+    if reduction == "mean":
+        res1 = res1.mean(dim=reduce_axes)
+        res2 = res2.mean(dim=reduce_axes)
```
Comment (edgarriba) [inline kornia/losses/total_variation.py:42]: can you extend with a small test for the different reduction cases ?

Comment (nitaifingerhut) [reply kornia/losses/total_variation.py:42]: sure

**Comment by nitaifingerhut** [comment]
@edgarriba added a reduction test
